btb/tuning/hyperparams/boolean.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+
+"""Package where the boolean hyperparameters are defined."""
+
+import numpy as np
+
+from btb.tuning.hyperparams.base import BaseHyperParam
+
+
+class BooleanHyperParam(BaseHyperParam):
+    """Boolean Hyperparameter Class.
+
+    The boolean hyperparameter class it is responsible for the transformation of boolean values in
+    to normalized search space of [0, 1]^K and also providing samples of those or the inverse
+    transformation from search space to hyperparameter space.
+
+    Hyperparameter space:
+        ``{True, False}``
+
+    Search space:
+        ``{0, 1}^K``
+
+    Attributes:
+        K (int):
+            Number of dimensions that this hyperparameter uses to be represented in the search
+            space.
+    """
+
+    K = 1
+
+    def _within_hyperparam_space(self, values):
+        if values.dtype is not np.dtype('bool'):
+            raise ValueError('Value not within hyperparameter range.')
+
+    def _inverse_transform(self, values):
+        """Inverse trasnform one or more values from search space [0, 1]^K.
+
+        Converts normalized ``values`` from the search space [0, 1]^K to the original space
+        of ``True`` and ``False`` by casting those values to ``boolean``.
+
+        Args:
+            values (ArrayLike):
+                Single value or 2D ArrayLike of normalized values.
+
+        Returns:
+            denormalized (Union[object, List[object]]):
+                Denormalized value or list of denormalized values.
+
+        Examples:
+            >>> instance = BooleanHyperParam()
+            >>> instance.inverse_transform([0, 1])
```

Comment:
The example snippets should be using `_inverse_transform` instead of `inverse_transform`.

Also, let's show here only the usage when values is a 2D array, as a 1D array should never reach this method.